### PR TITLE
detect Vimball (.vba, .vmb) files as Vim script

### LINKED ISSUE #501: .vba is not always Visual Basic
If you look at the percentage of language of the project chrisbra/Recover.vim...

https://github.com/chrisbra/Recover.vim

...you will see that it is reported as about 47% "Visual Basic".  I presume this is because the project has files with the .vba extension.  FYI, that extension here is being used to mean VimBall Archiver, which has nothing whatsoever to do with Visual Basic.

### PR DESCRIPTION
**Pull request by smola**
Fixes #501

## Description

- added .vba and .vmb extensions to Vim script
- added .vba disambiguation
- added Vimball sample with .vba extension (Vim license) from https://github.com/nelstrom/vim-textobj-rubyblock/blob/master/textobj-rubyblock.vba
- added Vimball sample with .vmb extension (WTFPL) from https://github.com/CDSoft/todo/blob/master/todo.vmb
- added Visual Basic sample with .vba extension (MIT license) from https://github.com/Akaizoku/concatenate/blob/master/procedures.vba

## Checklist:
- [x] **I am associating a language with a new file extension.**
  - [x] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      - https://github.com/search?q=extension%3Avba+usevimball+NOT+nothack&type=Code
      - https://github.com/search?q=extension%3Avmb+usevimball+NOT+nothack&type=Code
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/nelstrom/vim-textobj-rubyblock/blob/master/textobj-rubyblock.vba
    - Sample license(s): Vim License
  - [x] I have included a change to the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

File: samples/Vim script/textobj-rubyblock.vba [hunk 1]
```diff
@@ -0,0 +1,216 @@
+" Vimball Archiver by Charles E. Campbell, Jr., Ph.D.
```
Comment (pchaigno) [inline samples/Vim script/textobj-rubyblock.vba]: Could you also add an `.vmb` sample file?

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -5169,7 +5169,9 @@ Vim script:
   - viml
   - nvim
   extensions:
+  - ".vba"
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: The first extension should be the "primary extension", only subsequent extensions need to be sorted alphabetically. `.vim` is the primary extension for Vim scripts, no?

File: lib/linguist/heuristics.yml [hunk 3]
```diff
@@ -426,6 +426,11 @@ disambiguations:
     pattern: '^\s*(import.+(from\s+|require\()[''"]react|\/\/\/\s*<reference\s)'
   - language: XML
     pattern: '(?i:^\s*<\?xml\s+version)'
+- extensions: ['.vba']
+  rules:
+  - language: Vim script
+    pattern: '^UseVimball'
```
Comment (pchaigno) [inline lib/linguist/heuristics.yml]: Do you know if Vim scripts are invalid if they don't match this pattern?

File: test/test_heuristics.rb [hunk 4]
```diff
@@ -475,6 +475,13 @@ def test_tsx_by_heuristics
     })
   end
 
+  def test_vba_by_heuristics
+    assert_heuristics({
+      "Visual Basic" => all_fixtures("Visual Basic", "*.vba"),
+      "Vim script" => all_fixtures("XML", "*.vba")
```
Comment (pchaigno) [inline test/test_heuristics.rb]: Typo?
```suggestion
      "Vim script" => all_fixtures("Vim script", "*.vba")
```

Comment (smola) [reply test/test_heuristics.rb]: Yep. Thanks!

File: lib/linguist/languages.yml (hunk 2 above)
Comment (smola) [reply lib/linguist/languages.yml]: Indeed, `.vim` is still the primary extension. `.vba` and `.vmb` are Vimball archives (Vim plugins archived by [Vimball](https://www.vim.org/scripts/script.php?script_id=1502)). They are Vim scripts, but they will work only if the Vimball script is installed. They invoke it with `UseVimball` to extract the contents, which are in the rest of the archive.

AFAIK all Vimball archives have this `UseVimball` call. And if something has `.vba` and `.vmb` and it's not a Vimball archive, it's very unlikely that it's a different kind of Vim script.

File: lib/linguist/heuristics.yml (hunk 3 above)
Comment (smola) [reply lib/linguist/heuristics.yml]: Se above. Vim scripts will be valid if they don't match this, but then they'll have `.vim` extension.

Comment (pchaigno) [reply lib/linguist/heuristics.yml]: Good. Let's keep the heuristic as it is then.

**Pull request merged**